tests: sample.net.sockets.echo_server : frdm_k64f: build failure

**Describe the bug**

build failure.

<html>
<body>

CASE_NAME | ZEPHYR_VERSION | RESULT | PLATFORM | TEST_PATH
-- | -- | -- | -- | --
sample.net.sockets.echo_server | v3.7.0-rc1 | Build failure | frdm_k64f | samples/net/sockets/echo_server
sample.net.sockets.echo_server.802154.rf2xx.arduino | v3.7.0-rc1 | Build failure | frdm_k64f | samples/net/sockets/echo_server
sample.net.sockets.echo_server.usbnet | v3.7.0-rc1 | Build failure | frdm_k64f | samples/net/sockets/echo_server

</body>
</html>

Please also mention any information which could help others to understand
the problem you're facing:
 - What target platform are you using?
 frdm_k64f
 - What have you tried to diagnose or workaround this issue?
 git bisect
 - Is this a regression? If yes, have you been able to "git bisect" it to a
   specific commit?
 
```
cca390d24365e80ef274141151c53efe2b2e9af9
drivers: ethernet: eth_enet: add MMIO mapping

Adding MMIO mapping support in ENET module driver before MDIO, MAC
and PTP drivers initialization, so that they can reuse this mapping.

```

**To Reproduce**
Steps to reproduce the behavior:

scripts/twister -p frdm_k64f -T samples/net/sockets/echo_server 

**Expected behavior**
build pass

**Impact**

echo server

**Logs and console output**

```
/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/drivers/ethernet/nxp_enet/eth_nxp_enet.c: In function 'eth_nxp_enet_device_pm_action':
/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/drivers/ethernet/nxp_enet/eth_nxp_enet.c:787:34: error: 'const struct nxp_enet_mac_config' has no member named 'base'
  787 |                 ENET_Reset(config->base);
      |                                  ^~
/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/drivers/ethernet/nxp_enet/eth_nxp_enet.c:788:33: error: 'const struct nxp_enet_mac_config' has no member named 'base'
  788 |                 ENET_Down(config->base);
      |                                 ^~
[226/297] Building C object zephyr/drivers/gpio/CMakeFiles/drivers__gpio.dir/gpio_mcux.c.obj
```

**Environment (please complete the following information):**
 - OS: (e.g. Linux, MacOS, Windows)
 - Toolchain (e.g Zephyr SDK, ...)
 - Commit SHA or Version used: v3.7.0-rc1

--- a/drivers/ethernet/nxp_enet/eth_nxp_enet.c
+++ b/drivers/ethernet/nxp_enet/eth_nxp_enet.c
@@ -784,8 +784,8 @@ static int eth_nxp_enet_device_pm_action(const struct device *dev, enum pm_devic
 			return ret;
 		}
 
-		ENET_Reset(config->base);
-		ENET_Down(config->base);
+		ENET_Reset(data->base);
+		ENET_Down(data->base);
 		clock_control_off(config->clock_dev, (clock_control_subsys_t)config->clock_subsys);
 	} else if (action == PM_DEVICE_ACTION_RESUME) {
 		LOG_DBG("Resuming");

